extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -13,3 +13,121 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * @fileoverview Service for recaptcha components
+ * interacting with the 3p recaptcha bootstrap iframe
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {dev} from '../../../src/log';
+import {getIframe} from '../../../src/3p-frame';
+import {getService, registerServiceBuilder} from '../../../src/service';
+import {listenFor} from '../../../src/iframe-helper';
+import {removeElement} from '../../../src/dom';
+
+export class AmpRecaptchaService {
+
+  /**
+   * @param {!Window} window
+   */
+  constructor(window) {
+
+    /** @private {!Window} */
+    this.win_ = window;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {number} */
+    this.registeredElements_ = 0;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {!Deferred} */
+    this.willBeReady_ = new Deferred();
+  }
+
+  /**
+   * Function to register as a dependant of the AmpRecaptcha serivce.
+   * Used to create/destroy recaptcha boostrap iframe.
+   * @param {!AMP.BaseElement} elementImpl
+   * @return {Promise}
+   */
+  register(elementImpl) {
+    this.registeredElements_++;
+    if (!this.iframe_) {
+      this.initialize_(elementImpl);
+    }
+    return elementImpl.loadPromise(this.iframe_);
```

Comment:
instead of having each component create their own load promise and listen to the iframe's load event. Can we keep a reference to the iframe load event, and only return the reference to the promise.